python/MooseDocs/extensions/MooseObjectSyntax.py

```diff
@@ -0,0 +1,223 @@
+        self._child_objects = child_objects
+        self._root = root
+        self._repo = repo
+        self._name = None
+
+    def handleMatch(self, match):
+        """
+        Create a <p> tag containing the supplied description from the YAML dump.
+        """
+
+        # Extract match options and settings
+        action = match.group(2)
+        syntax = match.group(3)
+        settings, styles = self.getSettings(match.group(4))
+
+        # Locate description
+        node = self._yaml.find(syntax)
+        if not node:
+            return self.createErrorElement(message='Failed to locate {} syntax.'.format(syntax))
+
+        # Determine object name
+        self._name = node['name'].split('/')[-1]
+
+        if action == 'description':
+            el = self.descriptionElement(node, styles)
+        elif action == 'parameters':
+            el = self.parametersElement(node, styles)
+        elif action == 'inputfiles':
+            el = self.inputfilesElement(node, styles)
+        elif action == 'childobjects':
+            el = self.childobjectsElement(node, styles)
+        elif action == 'devel':
+            el = self.develElement(node, styles)
+        elif action == 'subobjects':
+            el = self.subobjectsElement(node, styles)
+        return el
+
+    def descriptionElement(self, node, styles):
+        """
+        Return the class description html element.
+
+        Args:
+            node[dict]: YAML data node.
+            styles[dict]: Styles from markdown.
+        """
+
+        if ('description' not in node) or (not node['description']):
+            return self.createErrorElement(message='Failed to locate class description for {} syntax.'.format(node['name']))
+
+        # Create the html element with supplied styles
+        el = self.addStyle(etree.Element('p'), **styles)
+        el.text = node['description']
+        return el
+
+    def parametersElement(self, node, styles):
+        """
+        Return table(s) of input parameters.
+
+        Args:
+            node[dict]: YAML data node.
+            styles[dict]: Styles from markdown.
+        """
+
+        # Create the tables (generate 'Required' and 'Optional' initially so that they come out in the proper order)
+        tables = collections.OrderedDict()
+        tables['Required'] = MooseDocs.MooseObjectParameterTable()
+        tables['Optional'] = MooseDocs.MooseObjectParameterTable()
+
+        # Loop through the parameters in yaml object
+        for param in node['parameters'] or []:
+            name = param['group_name']
+            if not name and param['required']:
+                name = 'Required'
+            elif not name and not param['required']:
+                name = 'Optional'
+
+            if name not in tables:
+                tables[name] = MooseDocs.MooseObjectParameterTable()
+            tables[name].addParam(param)
+
+        el = self.addStyle(etree.Element('div'), **styles)
```

Comment:
I have found passing styles to Element **and** any SubElement necessary to make it all work. This is because some elements allow certain CSS while others do not `<div>` v's  `<p>` for example. The text-alignment attribute being chief amongst them.

I have no elegant way to know which element one is attempting to modify of course... its on my todo list for a rainy day.

Just mentioning this here as an FYI!